wp-content/civi-extensions/goonjcustom/CRM/Goonjcustom/CivirulesAction/CreateEventForContact.php

```diff
@@ -52,7 +91,7 @@ public function processAction(CRM_Civirules_TriggerData_TriggerData $triggerData
 		}
 
 		$eventParams = [
-			'title' => 'Collection Camp',
+			'title' => $title,
```

Comment:
Move all the logic related to the generation of event code to a specific function and hence respect [SRP](https://en.wikipedia.org/wiki/Single-responsibility_principle).
```suggestion
			'title' => $this->getEventCode($createdDate, $address),
```